<?xml version="1.0"?>

<launch>

    <node name="rviz" pkg="rviz2" exec="rviz2" output="screen" args="-d $(find-pkg-share extract_line_segments)/launch/rviz.rviz">
        <param name="use_sim_time" value="true"/>
    </node>

    <executable cmd="ros2 bag play --clock 100 $(find-pkg-share extract_line_segments)/launch/run1" />

    <node name="extract_line_segments" pkg="extract_line_segments" exec="extract_line_segments_node" output="screen">
        <param name="douglas_peucker_distance" value="0.1"/>
        <param name="neighbor_distance" value="0.5"/>
        <param name="min_cluster_size" value="15"/>
        <param name="use_sim_time" value="true"/>
    </node>
</launch>